File: code-clone/src/app/shared/models/CloneData.ts
import {CloneFeedback} from "./CloneFeedback";

export class CloneData {
  cloneType: number;
  referenceLocation: [number, number];
  inputLocation: [number, number];
  feedback: CloneFeedback;
  referenceFileName: string;


  constructor(type: number, referenceLocation: [number, number], inputLocation: [number, number],referenceFileName?: string) {
    this.cloneType = type;
    this.referenceLocation = referenceLocation;
    this.inputLocation = inputLocation;
    this.referenceFileName = referenceFileName;
  }

  toString(): String {
    return "type: " + this.cloneType + ",\nref location: " + this.referenceLocation
      + ",\nreference file name: " + this.referenceFileName + ",\ninput location: " + this.inputLocation + ",\nfeedback: " + this.feedback;
  }

}

File: code-clone-detector/src/main/java/com/clonedetector/codeclonedetector/model/CloneData.java
package com.clonedetector.codeclonedetector.model;

import lombok.Data;
import org.springframework.data.annotation.Id;

import javax.persistence.Entity;
import java.util.Arrays;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "clones")
public class CloneData {

//    @Id
//    private long id;
    public int cloneType;
    public int[] referenceLocation;
    public int[] inputLocation;

    public CloneData(){

    }

    public String toString() {
        return "cloneType: " + cloneType + "\t Input Location: " + Arrays.toString(inputLocation) + "\tReference Location: " + Arrays.toString(referenceLocation);
    }
}
